src/codegen.cpp

```diff
@@ -2317,7 +2319,7 @@ static jl_cgval_t convert_julia_type(jl_codectx_t &ctx, const jl_cgval_t &v, jl_
                 new_tindex = ConstantInt::get(getInt8Ty(ctx.builder.getContext()), new_idx);
                 if (v.V && !v.ispointer()) {
                     // TODO: remove this branch once all consumers of v.TIndex understand how to handle a non-ispointer value
-                    return value_to_pointer(ctx, v.V, typ, new_tindex);
+                    return jl_cgval_t(value_to_pointer(ctx, v.V, v.typ), typ, new_tindex);
```

Comment:
ah, it wants the whole thing. But also, you should delete the unused parameter to value_to_pointer now also
```suggestion
                    return jl_cgval_t(value_to_pointer(ctx, v), typ, new_tindex);
```